### ISSUE TITLE
Crash when running on cluster with BoundServiceAccountTokenVolume feature gate enabled

### ISSUE THREAD

**Issue by javanthropus** [bitnami-labs/sealed-secrets#338]
The BoundServiceAccountTokenVolume feature gate modifies service account tokens to be provided via a projected volume.  Permissions are slightly different for the token file itself:

```
-rw-------    1 root     root          1058 Dec 31 14:31 /var/run/secrets/kubernetes.io/serviceaccount/..data/token
```

Normally, the token file is mode 644.  The difference results in a crash:

```
2019/12/31 14:37:26 Starting sealed-secrets controller version: v0.9.6+dirty
panic: open /var/run/secrets/kubernetes.io/serviceaccount/token: permission denied

goroutine 1 [running]:
main.main()
	/home/travis/gopath/src/github.com/bitnami-labs/sealed-secrets/cmd/controller/main.go:271 +0x215
```

The fix is to set `securityContext.fsGroup` for the podspec to something like the following:

```yaml
spec:
  securityContext:
    fsGroup: 65534
```

This results in the token file having mode 640 with a group owner of 65534.  Any `fsGroup` value is acceptable, but this one is typically the `nobody` group.

**Comment by mkmik** [comment]
Do you know from the top of your head in which versions such use of k8s `fsGroup` is supported?

**Comment by javanthropus** [comment]
It's part of the core/v1 spec:

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#podsecuritycontext-v1-core

That documentation is available down to version v1.10 at that site.

**Comment by mkmik** [comment]
ok; in any case, integration tests will check that on the supported k8s versions.

@javanthropus please review  #339
